static/js/publisher/builds/components/repoConnect.js

```diff
@@ -0,0 +1,348 @@
+
+    this.state = {
+      selectedRepo: "",
+      repoList: [],
+      selectedOrganization: "",
+      organizations: this.props.organizations,
+      user: this.props.user,
+      isRepoListDisabled: true,
+      message: { icon: "", description: "" },
+      snapName: this.props.snapName,
+      yamlFilePath: ""
+    };
+
+    this.handleRepoSelect = this.handleRepoSelect.bind(this);
+    this.handleOrganiationSelect = this.handleOrganiationSelect.bind(this);
+    this.fetchRepoList = this.fetchRepoList.bind(this);
+    this.handleRefreshButtonClick = this.handleRefreshButtonClick.bind(this);
+  }
+
+  sortByValue(a, b) {
+    return a.value.localeCompare(b.value);
+  }
+
+  /**
+   * Handle repository select event
+   *
+   * @param repository
+   */
+  handleRepoSelect(selectedRepo) {
+    this.setState(
+      {
+        selectedRepo: selectedRepo,
+        message: { icon: "", description: "" }
+      },
+      () => this.checkRepo(selectedRepo)
+    );
+  }
+
+  /**
+   * Handle organization select event
+   *
+   * @param organization
+   */
+  handleOrganiationSelect(selectedOrganization) {
+    this.setState({ selectedOrganization: selectedOrganization }, () =>
+      this.fetchRepoList()
+    );
+  }
+
+  /**
+   * Fetch repo list of the selected organization
+   *
+   */
+  fetchRepoList() {
+    const { selectedOrganization, user } = this.state;
+    let url = "";
+
+    if (selectedOrganization === user.login) {
+      url = "/publisher/github/get-repos";
+    } else {
+      url = `/publisher/github/get-repos?org=${selectedOrganization}`;
+    }
+    this.setState({
+      isRepoListDisabled: true,
+      message: { icon: "spinner", description: "" }
+    });
+
+    fetch(url)
+      .then(res => res.json())
+      .then(result => {
+        let newRepoList = result
+          .map(el => {
+            return { value: el.name };
+          })
+          .sort(this.sortByValue);
+        this.setState({
+          repoList: newRepoList,
+          isRepoListDisabled: false,
+          message: { icon: "", description: "" }
+        });
+      })
+      .catch(() => {
+        this.setState({
+          isRepoListDisabled: false,
+          message: {
+            icon: "error",
+            description:
+              "There was an error fetching your repository list. Please try again."
+          }
+        });
+      });
+  }
+
+  /**
+   * Check if repository is suitable for buld
+   *
+   * @param selectedRepo
+   */
+  checkRepo(selectedRepo) {
+    const { snapName, repoList, selectedOrganization } = this.state;
+    if (selectedRepo && repoList.some(el => el.value === selectedRepo)) {
+      const url = `/${snapName}/builds/validate-repo?repo=${selectedOrganization}/${selectedRepo}`;
+
+      this.setState({
+        isRepoListDisabled: true,
+        message: { icon: "spinner", description: "" }
+      });
+
+      fetch(url)
+        .then(res => res.json())
+        .then(result => {
+          this.setState({
```

Comment:
Does the API return a result.error.type or some such that you can match against for NAME_MISMATCH so you don't have to string match later on? The result.error.message could change